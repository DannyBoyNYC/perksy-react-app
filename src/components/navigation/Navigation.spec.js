const React = require('react');
const PropTypes = require('prop-types');

const Navigation = require('./index').default;

describe('<Navigation.* />', () => {
  let component;
  const requiredProps = {
    links: [
      {
        text: 'Foo',
        url: '#'
      },
      {
        text: 'Bar',
        url: '#'
      }
    ]
  };

  describe('<Navigation.Horizontal />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Navigation.Horizontal {...requiredProps} />,
        'Navigation'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a link for each item', () => {
      const options = {
        context: {
          router: {
            prefetch: () => {},
            route: '/mock-route',
            pathname: 'mock-path'
          }
        },
        childContextTypes: {
          router: PropTypes.object
        }
      };
      component = mountWithTheme(
        <Navigation.Horizontal {...requiredProps} />,
        options
      );
      expect(component).toContainMatchingElements(2, 'a');
    });
  });

  describe('<Navigation.Vertical />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Navigation.Vertical {...requiredProps} />,
        'Navigation'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
