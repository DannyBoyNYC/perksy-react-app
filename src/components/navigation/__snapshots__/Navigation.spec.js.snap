// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation.* /> <Navigation.Horizontal /> should render correctly with all required props 1`] = `
<nav
  className="Navigation-root-0-1-1 Navigation-horizontal-0-1-3"
>
  <Unordered
    className="Navigation-list-0-1-2"
    items={
      Array [
        Object {
          "text": "Foo",
          "url": "#",
        },
        Object {
          "text": "Bar",
          "url": "#",
        },
      ]
    }
    onPrerender={[Function]}
  />
</nav>
`;

exports[`<Navigation.* /> <Navigation.Vertical /> should render correctly with all required props 1`] = `
<nav
  className="Navigation-root-0-1-1 Navigation-vertical-0-1-4"
>
  <Unordered
    className="Navigation-list-0-1-2"
    items={
      Array [
        Object {
          "text": "Foo",
          "url": "#",
        },
        Object {
          "text": "Bar",
          "url": "#",
        },
      ]
    }
    onPrerender={[Function]}
  />
</nav>
`;
